### ISSUE TITLE
Grid floating incorrectly

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#646]
Grid is not floating correctly in the form:

Example has the following HTML Structure:
``` html
<div class="grid">
    <div class="grid-row">
        <div class="grid-col-9 floating form-group" />
        <div class="grid-col-3 floating form-group" />
        <div class="grid-col-9 floating form-group" />
        <div class="grid-col-3 floating form-group" />
    </div>
</div>
```
Example:
![bildschirmfoto 2014-12-10 um 17 59 12](https://cloud.githubusercontent.com/assets/1698337/5380176/6ef788f0-8096-11e4-9434-612d0596d30c.png)

**Comment by alexander-schranz** [comment]
<div class="grid">
    <div class="grid-row">
        <div class="grid-col-6 floating form-group" />
        <div class="grid-col-3 floating form-group" />
        <div class="grid-col-6 floating form-group" />
        <div class="grid-col-3 floating form-group" />
    </div>
</div>

currently seems the grid dont work with multiline.

![bildschirmfoto 2015-03-11 um 16 00 52](https://cloud.githubusercontent.com/assets/1698337/6599055/d8301004-c807-11e4-9df1-f61e0565072c.png)

the grid system should have NO selector on `first-child` or `.grid-col-3.floating:nth-child(4n+1)` thats very EVIL the padding-left must never be set to 0 for a specific combincation. if the html structure is done correctly and maybe add a class `no-padding` for some grid items will make the grid float correctly.

The `.custom-checkbox` container should also have the same height as the textfield
``` css
.custom-checkbox {
    top: 0;
    height: 30px;
    .icon,
    input  {
        top: 7px;
    }
}
```

**Comment by ClockRight** [comment]
Same issue with this block:
```
<block name="blocks" default-type="editor_image">
                    <meta>
                        <title lang="de">Inhalte</title>
                        <title lang="en">Content</title>
                    </meta>
                    <types>
                       <type name="editor_image">
                            <meta>
                                <title lang="de">Editor mit Bild</title>
                                <title lang="en">Editor with image</title>
                            </meta>
                            <properties>
                                <property name="title" type="text_line" colspan="9">
                                    <meta>
                                        <title lang="de">Titel</title>
                                        <title lang="en">Title</title>
                                    </meta>
                                </property>
                                <property name="hide" type="checkbox" colspan="3">
                                    <meta>
                                        <title lang="de">Verbergen</title>
                                        <title lang="en">Hide</title>
                                    </meta>
                                </property>
                                <property name="images" type="media_selection" colspan="3">
                                    <meta>
                                        <title lang="de">Bilder</title>
                                        <title lang="en">Images</title>
                                    </meta>
                                    <params>
                                        <param name="type" value="image"/>
                                        <param name="displayOptions" type="collection">
                                            <param name="leftTop" value="false"/>
                                            <param name="top" value="true"/>
                                            <param name="rightTop" value="false"/>
                                            <param name="left" value="true"/>
                                            <param name="middle" value="false"/>
                                            <param name="right" value="true"/>
                                            <param name="leftBottom" value="false"/>
                                            <param name="bottom" value="true"/>
                                            <param name="rightBottom" value="false"/>
                                        </param>
                                    </params>
                                </property>
                                <property name="article" type="text_editor" colspan="9">
                                    <meta>
                                        <title lang="de">Artikel</title>
                                        <title lang="en">Article</title>
                                    </meta>
                                </property>

                            </properties>
                        </type>
                    </types>
                </block>
```

![block_error](https://cloud.githubusercontent.com/assets/17072809/13730716/28395486-e957-11e5-9612-73057ab77ff9.png)

**Comment by tobias47n9e** [comment]
Not sure if related, but after a Textarea, the grid seems to have a bug too:

![pasted image at 2016_04_07 15_49](https://cloud.githubusercontent.com/assets/6953323/14353471/7d81741a-fcd9-11e5-87ad-8c04b9308cf4.png)

![pasted image at 2016_04_07 15_50](https://cloud.githubusercontent.com/assets/6953323/14353477/84d77ea8-fcd9-11e5-8bac-7064f667c77f.png)

**Comment by chirimoya** [comment]
![image](https://cloud.githubusercontent.com/assets/627313/14910456/ab9afcac-0ded-11e6-9e96-86ccb3e8d45e.png)

**Comment by c00n84** [comment]
![image](https://cloud.githubusercontent.com/assets/841720/14910464/c8a40c62-0ded-11e6-8b1c-2549d8ae8926.png)

**Comment by ReneZumtobel** [comment]
I talked with @marcelmoosbrugger  about this problem and we made some changes in PR https://github.com/sulu/sulu/pull/2900 but the problem would be solved if you don´t use the grid like this: 
```
<div class="grid">
    <div class="grid-row">
        <div class="grid-col-9 floating form-group" />
        <div class="grid-col-3 floating form-group" />
        <div class="grid-col-9 floating form-group" />
        <div class="grid-col-3 floating form-group" />
    </div>
</div>
```

But use it like this instead:

```
<div class="grid">
    <div class="grid-row">
        <div class="grid-col-9 floating form-group" />
        <div class="grid-col-3 floating form-group" />
    </div>
    <div class="grid-row">
        <div class="grid-col-9 floating form-group" />
        <div class="grid-col-3 floating form-group" />
    </div>
</div>
```

**Comment by danrot** [comment]
Fixed by #2900
